### ISSUE TITLE
Keep "withKeycloak" HOC for those using React Class-Based Components

### ISSUE THREAD

**Issue by ivangfr** [react-keycloak/react-keycloak#111]
Hi guys, I've tried to contact you via `Gitter` about a question I have, but no response so far. That is why I am trying now via `Github` issue.

I've recently update my application (https://github.com/ivangfr/springboot-react-keycloak) to `@react-keycloak/web` from `2.1.1` to `3.0.0`. There are some breaking changes very well explained at https://github.com/react-keycloak/react-keycloak/blob/master/packages/web/MIGRATION.md.

It seems that you are not going to support `withKeycloak` HOC anymore. So, in this case, we need to migrate all our **React Class-Based Components** to **React Hooks**, right?

As I am not expert in React, I am here to ask your opinion: instead of refactoring all my Class-Based Components to Hooks, I've implemented the React Function below. Is it a good approach?
```
import React from 'react'
import { useKeycloak } from '@react-keycloak/web'

function withKeycloak(Component) {
  return function WrappedComponent(props) {
    const { keycloak, initialized } = useKeycloak()
    return <Component {...props} keycloakInitialized={initialized} keycloak={keycloak} />
  }
}

export default withKeycloak
```

Btw, would be a good idea to keep `withKeycloak` HOC for those using React Class-Based Components?

Best regards!

**Comment by panz3r** [comment]
Hi @ivangfr ,

Thanks for your feedback and contribution.

The `withKeycloak` HOC as been re-implemented for the following modules version:
 - `@react-keycloak/ssr@3.3.0`
 - `@react-keycloak/web@3.4.0`

**Comment by tiaanduplessis** [comment]
@panz3r I think a note should be made in the [migration guide](https://github.com/react-keycloak/react-keycloak/blob/master/packages/ssr/MIGRATION.md) that the HOC has been re-implemented in `@react-keycloak/ssr@3.3.0` and `@react-keycloak/web@3.4.0`. Would be happy to create a PR.
